File: app/src/main/java/com/mgg/callbackhandler/LibCheckerApp.java
package com.mgg.callbackhandler;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import timber.log.Timber;

public class LibCheckerApp extends Application {
    static LibCheckerApp instancesLibCheckerApp = null;

    public static LibCheckerApp getInstance() {
        if (instancesLibCheckerApp == null) {
            throw new IllegalStateException("LibCheckerApp does not exist");
        }
        return instancesLibCheckerApp;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        JNIUtils.setClassLoader(this.getClassLoader());
        Timber.plant(new Timber.Tree() {
            @Override
            protected void log(int i, @Nullable String s, @NonNull String s1, @Nullable Throwable throwable) {
            }
        });
        instancesLibCheckerApp = this;
    }
}

File: app/src/main/java/com/mgg/callbackhandler/LibCheckerApp.kt
package com.mgg.callbackhandler

import android.app.Application
import timber.log.Timber

class LibCheckerApp : Application() {

    override fun onCreate() {
        super.onCreate()
        JNIUtils.setClassLoader(this.classLoader)
        Timber.plant(Timber.DebugTree())
        app = this
    }

    companion object {
        //noinspection StaticFieldLeak
        lateinit var app: Application
    }
}